api/server/router/container/backend.go

```diff
@@ -43,7 +43,7 @@ type stateBackend interface {
 	ContainerStop(name string, seconds int) error
 	ContainerUnpause(name string) error
 	ContainerUpdate(name string, hostConfig *container.HostConfig) ([]string, error)
-	ContainerWait(name string, timeout time.Duration) (int, error)
+	ContainerWait(name string, timeout time.Duration, waitForRemoval bool) (int, error)
```

Comment:
In that case, the only other options I'm seeing is:

Listening for events with filter "container=<id>,action=die" to get the exitcode and an added "action=destroy" if `--rm` was specified? 

If we're going to be watching events we may as well do it all there (didn't realize that we would still need to do so because of the race between start and wait when I proposed my solution).

That would actually remove the need to have different code path depending on whether with have a tty or not.